### PR TITLE
[20131] Update TypeInformationParameter to contain xtype

### PR DESCRIPTION
**Pull request by adriancampo**
## Description

This PR depends on #4179

## Contributor Checklist
- [X] Commit messages follow the project guidelines. 
- [X] The code follows the style guidelines of this project. 
- N/A Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- N/A Any new/modified methods have been properly documented using Doxygen. 
- N/A Changes are ABI compatible. 
- N/A Changes are API compatible. 
- N/A New feature has been added to the `versions.md` file (if applicable).
- N/A New feature has been documented/Current behavior is correctly described in the documentation. 
    
- N/A Applicable backports have been included in the description.

## Reviewer Checklist
- [ ] The PR has a milestone assigned.
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: include/fastdds/dds/core/policy/QosPolicies.hpp [hunk 1]
```diff
@@ -23,6 +23,7 @@
 #include <vector>
 
 #include <fastdds/dds/core/policy/ParameterTypes.hpp>
+#include <fastdds/dds/xtypes/type_representation/detail/dds_xtypes_typeobject.hpp>
```
Comment (JLBuenoLopez-eProsima) [inline include/fastdds/dds/core/policy/QosPolicies.hpp:26]: The idea is to include `fastdds/dds/xtypes/type_representation/TypeObject.hpp` that includes this header.

File: include/fastdds/dds/core/policy/QosPolicies.hpp [hunk 2]
```diff
@@ -23,6 +23,7 @@
...
 #include <fastdds/rtps/attributes/ExternalLocators.hpp>
 #include <fastdds/rtps/attributes/PropertyPolicy.h>
 #include <fastdds/rtps/attributes/RTPSParticipantAllocationAttributes.hpp>
```
Comment (JLBuenoLopez-eProsima) [inline include/fastdds/dds/core/policy/QosPolicies.hpp:29]: To ensure that the migration has been correctly completed, please remove the old header `fastrtps/types/TypeObject.h`.

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]
Two nitpicks

**Comment by JLBuenoLopez-eProsima** [comment]
@adriancampo take a look because the compilation is failing:

```
In file included from /home/runner/work/Fast-DDS/Fast-DDS/src/fastrtps/src/cpp/statistics/fastdds/domain/DomainParticipantImpl.hpp:39,
                 from /home/runner/work/Fast-DDS/Fast-DDS/src/fastrtps/test/unittest/statistics/dds/StatisticsDomainParticipantMockTests.cpp:25:
/home/runner/work/Fast-DDS/Fast-DDS/src/fastrtps/test/mock/dds/DomainParticipantImpl/fastdds/domain/DomainParticipantImpl.hpp: In member function ‘eprosima::fastrtps::rtps::SampleIdentity eprosima::fastdds::dds::DomainParticipantImpl::get_type_dependencies(const TypeIdentifierSeq&) const’:
/home/runner/work/Fast-DDS/Fast-DDS/src/fastrtps/test/mock/dds/DomainParticipantImpl/fastdds/domain/DomainParticipantImpl.hpp:582:35: error: ‘class eprosima::fastrtps::rtps::RTPSParticipant’ has no member named ‘typelookup_manager’
  582 |         return rtps_participant_->typelookup_manager()->get_type_dependencies(in);
      |                                   ^~~~~~~~~~~~~~~~~~
/home/runner/work/Fast-DDS/Fast-DDS/src/fastrtps/test/mock/dds/DomainParticipantImpl/fastdds/domain/DomainParticipantImpl.hpp: In member function ‘eprosima::fastrtps::rtps::SampleIdentity eprosima::fastdds::dds::DomainParticipantImpl::get_types(const TypeIdentifierSeq&) const’:
/home/runner/work/Fast-DDS/Fast-DDS/src/fastrtps/test/mock/dds/DomainParticipantImpl/fastdds/domain/DomainParticipantImpl.hpp:588:35: error: ‘class eprosima::fastrtps::rtps::RTPSParticipant’ has no member named ‘typelookup_manager’
  588 |         return rtps_participant_->typelookup_manager()->get_types(in);
      |                                   ^~~~~~~~~~~~~~~~~~
```

It seems to be caused by one mock class.

File: include/fastdds/dds/core/policy/QosPolicies.hpp (hunk 1 above)
Comment (adriancampo) [reply include/fastdds/dds/core/policy/QosPolicies.hpp:26]: Removed in [83756a6](https://github.com/eProsima/Fast-DDS/pull/4140/commits/83756a6445a363860a622ce24e1cbea2985a6c6a)

File: include/fastdds/dds/core/policy/QosPolicies.hpp (hunk 2 above)
Comment (adriancampo) [reply include/fastdds/dds/core/policy/QosPolicies.hpp:29]: Removed in [83756a6](https://github.com/eProsima/Fast-DDS/pull/4140/commits/83756a6445a363860a622ce24e1cbea2985a6c6a)

**Comment by adriancampo** [comment]
@richiprosima please test this

**Comment by adriancampo** [comment]
@richiprosima please test this

File: include/fastdds/dds/core/policy/QosPolicies.hpp (hunk 1 above)
Comment (adriancampo) [reply include/fastdds/dds/core/policy/QosPolicies.hpp:27]: That would create a circular dependency with dds_xtypes_typeobjectPubSubTypes.h
An ugly one :S

**Comment by adriancampo** [comment]
@richiprosima please test this

**Comment by adriancampo** [comment]
@richiprosima please test this

**Comment by adriancampo** [comment]
@richiprosima please test this

File: include/fastdds/dds/core/policy/QosPolicies.hpp [hunk 3]
```diff
@@ -34,13 +34,14 @@
...
 #include <fastdds/rtps/resources/ResourceManagement.h>
-#include <fastrtps/types/TypeObject.h>
 #include <fastrtps/utils/collections/ResourceLimitedVector.hpp>
 
 namespace eprosima {
 namespace fastdds {
 namespace dds {
 
+using DomainId_t = uint32_t;
+
```
Comment (JLBuenoLopez-eProsima) [inline include/fastdds/dds/core/policy/QosPolicies.hpp:45]: This is going to be defined in another file. A PR is going to be issued creating this file. This PR should wait until that PR is merged.

File: include/fastdds/dds/xtypes/type_representation/ITypeObjectRegistry.hpp [hunk 4]
```diff
@@ -18,7 +18,7 @@
  */
 
 #include <fastdds/dds/xtypes/type_representation/TypeObject.hpp>
-
+#include <fastdds/dds/core/ReturnCode.hpp>
```
Comment (JLBuenoLopez-eProsima) [inline include/fastdds/dds/xtypes/type_representation/ITypeObjectRegistry.hpp:21]: Please, sort alphabetically.

I am also missing the `<string>` header

File: test/xtypes/XTypesTests.cpp [hunk 5]
```diff
@@ -1121,189 +1120,6 @@ TEST_F(xtypestests, DISABLED_TypeIdentifierNamesIgnored)
     pub.waitDiscovery(true, 3);
 }
 
-/**** TYPE INFORMATION ****/
-
-/*
- * Both endpoints share the same type using XCDR1 so they must match.
- * This represents classical match.
- */
-TEST_F(xtypestests, TypeInformationSameType)
```
Comment (JLBuenoLopez-eProsima) [inline test/xtypes/XTypesTests.cpp:1129]: Please, ensure that these same test cases are tested in the blackbox/communication tests.

File: include/fastdds/dds/core/policy/QosPolicies.hpp (hunk 3 above)
Comment (adriancampo) [reply include/fastdds/dds/core/policy/QosPolicies.hpp]: Waiting for https://github.com/eProsima/Fast-DDS/pull/4179

File: include/fastdds/dds/xtypes/type_representation/ITypeObjectRegistry.hpp (hunk 4 above)
Comment (adriancampo) [reply include/fastdds/dds/xtypes/type_representation/ITypeObjectRegistry.hpp]: Fixed in [b660c7e](https://github.com/eProsima/Fast-DDS/pull/4140/commits/b660c7e19c132624708e6e06a7b989926917aff7)

File: test/xtypes/XTypesTests.cpp (hunk 5 above)
Comment (adriancampo) [reply test/xtypes/XTypesTests.cpp:1129]: Will create a PR with tests

**Comment by adriancampo** [comment]
@richiprosima please test this

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM with green CI (build)

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

**Pull request merged**